src/psygnal/_signal.py

```diff
@@ -1280,49 +1277,47 @@ def __call__(self, args: tuple[object, ...]) -> bool:
     def __eq__(self, other: object) -> bool:
         return (
             isinstance(other, _BoundMethodCaller)
-            and self._ref == other._ref
-            and self._method_name == other._method_name
+            and self._method_ref == other._method_ref
```

Comment:
As I check this comparison always returns false if the object is removed. I do not find this in the documentation, but I'm not sure if it was important in this context.

Reply:
thanks for having a look!  let me give that a little extra consideration.

Reply:
interesting... yeah, that _is_ different from the equality of a `weakref.ref` to an object (that comparison is `True` both before and after object deletion).   annoying :joy:

Reply:
I added some more tests, and fixed it (by also storing an object ref for the purpose of equality)... will merge once tests pass